### DISCUSSION TITLE
Error: A query attribute must be specified and must be a string. What am I doing wrong?

### DISCUSSION THREAD

**Discussion by GuqinLing** [community/community#24865] [Apps, API and Webhooks]
<p>when i use github api v4 with curl in linux,  error meesages as follows:</p>
<pre><code>curl -X POST -H "Authorization: token 7d93b8f3e7********7decfca090d8c3" -d '{"mutation":"mutation {addReaction(input:{subjectId:\"MDU6SXNzd*********yMjE=\"},content:HOORAY){reaction{content},subject{id}}}"}' https://api.github.com/graphql
</code></pre>
<p>{“errors”:[{“message”:“A query attribute must be specified and must be a string.”}]}.</p>
<p>I verify my token and subjectId in <a href="https://developer.github.com/v4/explorer/" rel="nofollow noopener">https://developer.github.com/v4/explorer/</a>  ,it’s ok.</p>
<p>pls tell me what’s wrong with me .thanks</p>

**Comment by GuqinLing** [comment, marked as answer]
<p>I solved this problem.</p>
<p>I use the “query” reply the first “mutation”</p>

**Comment by krasimir** [comment]
<p><a class="mention" href="/guqinling">@guqinling</a> can you post your solution.</p>

**Comment by paddyfeerick** [comment]
<p>In other words use:</p>
<ul>
<li>-d <strong>{“query”</strong> : "mutation {…</li>
</ul>
<p>instead of </p>
<ul>
<li>-d { <strong>“mutation”</strong> : "mutation {</li>
</ul>
<p>I was banging my head on the wall too…</p>
